it/src/main/java/com/google/cloud/teleport/it/testcontainers/TestContainerResourceManager.java

```diff
@@ -0,0 +1,184 @@
+import org.testcontainers.containers.GenericContainer;
+
+/** Interface for managing TestContainers resources in integration tests. */
+public abstract class TestContainerResourceManager<T extends GenericContainer<?>> {
+  private static final Logger LOG = LoggerFactory.getLogger(TestContainerResourceManager.class);
+
+  private final T container;
+  private final boolean usingStaticContainer;
+  private final String host;
+  private final int port;
+
+  protected <B extends TestContainerResourceManager.Builder<?>> TestContainerResourceManager(
+      T container, B builder) {
+    this.container = container;
+    this.usingStaticContainer = builder.useStaticContainer;
+    this.host = builder.host;
+    this.port = builder.port;
+
+    if (!usingStaticContainer) {
+      container.start();
+    } else if (this.host == null || this.port < 0) {
+      throw new TestContainerResourceManagerException(
+          "This manager was configured to use a static resource, but the host and port were not properly set.");
+    }
+  }
+
+  /**
+   * Returns the host that the resource is running on. If a host was specified in the builder, then
+   * this method will return that value. Otherwise, the host will default to localhost.
+   *
+   * @return the host that the resource is running on
+   */
+  protected String getHost() {
+    if (host == null) {
+      return container.getHost();
+    }
+    return host;
+  }
+
+  /**
+   * Returns the port that the resource is actually mapped to on the host. When using a non-static
+   * resource, the TestContainers framework will map the resource port to a random port on the host.
+   * This method will return the mapped port that the resource traffic is routed through.
+   *
+   * <p>Note: When using a static container, the port that was specified in the builder will be
+   * returned regardless of what is passed into this method as a parameter.
+   *
+   * @param mappedPort the port the resource was configured to listen on
+   * @return the mapped port on the host
+   */
+  protected int getPort(int mappedPort) {
+    if (port < 0) {
+      return container.getMappedPort(mappedPort);
+    }
+    return port;
+  }
+
+  /**
+   * Deletes all created resources (VM's, etc.) and stops the container, making the manager object
+   * unusable.
+   *
+   * @throws TestContainerResourceManagerException if there is an error deleting the TestContainers
+   *     resources.
+   */
+  protected boolean cleanupAll() {
+    LOG.info("Attempting to cleanup TestContainers manager.");
+
+    if (!usingStaticContainer) {
+      try {
+        container.close();
+      } catch (Exception e) {
+        LOG.error("Failed to close TestContainer resources. {}", e.getMessage());
+        return false;
+      }
+    } else {
+      LOG.info("This manager was configured to use a static resource that will not be cleaned up.");
+    }
+
+    LOG.info("TestContainers manager successfully cleaned up.");
+    return true;
+  }
```

Comment:
nit / might be 100% preference based but the code would look cleaner short circuiting on `usingStaticContainer` e.g.,

```suggestion
  protected boolean cleanupAll() {
    if (usingStaticContainer) {
       LOG.info("This manager was configured to use a static resource that will not be cleaned up.");
       return true;
    }
    
    LOG.info("Attempting to cleanup TestContainers manager.");
    try {
      container.close();
      LOG.info("TestContainers manager successfully cleaned up.");
      return true;
    } catch (Exception e) {
      LOG.error("Failed to close TestContainer resources. {}", e.getMessage());
      return false;
    }
  }
```

Reply:
I agree.